src/main/java/edu/cornell/kfs/concur/batch/service/impl/ConcurStandardAccountExtractPdpEntryServiceImpl.java

```diff
@@ -162,16 +167,121 @@ public Integer getPayeeNameFieldSize() {
-            LOG.debug("getConcurParamterValue, parameterName: " + parameterName + " paramterValue: " + parameterValue);
-        }
         return parameterValue;
     }
 
     public void setPayeeNameFieldSize(Integer payeeNameFieldSize) {
         this.payeeNameFieldSize = payeeNameFieldSize;
     }
     
+    @Override
+    public PdpFeedFileBaseEntry removeNonReimbursableSectionsFromPdpFeedFileBaseEntry(PdpFeedFileBaseEntry pdpFeedFileBaseEntry, 
+            ConcurStandardAccountingExtractBatchReportData reportData) {
+        LOG.debug("Entering removeNonReimbursableSectionsFromPdpFeedFileBaseEntry");
+        PdpFeedFileBaseEntry newBaseEntry = new PdpFeedFileBaseEntry();
+        newBaseEntry.setHeader(copyHeaderEntry(pdpFeedFileBaseEntry.getHeader()));
+        newBaseEntry.setVersion(pdpFeedFileBaseEntry.getVersion());
+        addGroupEntriesToNewBaseEntry(newBaseEntry, pdpFeedFileBaseEntry);
+        newBaseEntry.setTrailer(buildPdpFeedTrailerEntry(newBaseEntry, reportData));
+        return newBaseEntry;
+    }
+
+    private void addGroupEntriesToNewBaseEntry(PdpFeedFileBaseEntry newBaseEntry, PdpFeedFileBaseEntry originalBaseEntry) {
+        for (PdpFeedGroupEntry originalGroupEntry : originalBaseEntry.getGroup()) {
+            PdpFeedGroupEntry newGroupEntry = copyGroupEntry(originalGroupEntry);
+            addDetailEntriesToNewGroupEntry(newGroupEntry, originalGroupEntry);
+            if (CollectionUtils.isNotEmpty(newGroupEntry.getDetail())) {
+                newBaseEntry.getGroup().add(newGroupEntry);
+            } else {
+                LOG.debug("addGroupEntriesToNewBaseEntry, not adding group for" + newGroupEntry.getPayeeName());
+            }
+        }
+    }
+    
+    private void addDetailEntriesToNewGroupEntry(PdpFeedGroupEntry newGroupEntry, PdpFeedGroupEntry originalGroupEntry) {
+        for (PdpFeedDetailEntry originalDetailEntry : originalGroupEntry.getDetail()) {
+            PdpFeedDetailEntry newDetailEntry = copyDetailEntry(originalDetailEntry);
+            KualiDecimal originalDetailTransactionTotal = KualiDecimal.ZERO;
+            for (PdpFeedAccountingEntry originalAccountingEntry : originalDetailEntry.getAccounting()) {
+                originalDetailTransactionTotal = originalDetailTransactionTotal.add(originalAccountingEntry.getAmount());
+                if (originalAccountingEntry.getAmount().isPositive()) {
+                    newDetailEntry.getAccounting().add(copyAccountingEntry(originalAccountingEntry));
+                } else {
+                    LOG.debug("addDetailEntriesToNewGroupEntry, not adding accounting entry: " + originalAccountingEntry.toString());
+                }
+            }
+            addNewPaymentDetailToGroup(newGroupEntry, newDetailEntry, originalDetailTransactionTotal);
+        }
+    }
+
+    private void addNewPaymentDetailToGroup(PdpFeedGroupEntry newGroupEntry, PdpFeedDetailEntry newDetailEntry, KualiDecimal originalDetailTransactionTotal) {
```

Comment:
Or, since this method is only adding up to one payment detail at a time, perhaps it would be better to rename it to "addNewPaymentDetailToGroupIfTotalIsPositive"